r1cs/gadgets/std/src/groups/nonnative/mod.rs

```diff
@@ -0,0 +1,97 @@
+pub mod nonnative_group_gadget;
+
+use crate::prelude::*;
+use algebra::{Field, Group};
+use r1cs_core::{ConstraintSystem, SynthesisError};
+
+use std::{borrow::Borrow, fmt::Debug};
+
+pub trait NonNativeGroupGadget<G: Group, ConstraintF: Field, SimulationF: Field>:
```

Comment:
I don't think this is needed. We should do like NonNativeFieldGadget being a simple struct implementing FieldGadget. I know that you had problems doing in this way, but let's have a call to understand what they are and how to fix them if possible

Reply:
Done